File: CS/App_Code/Device.cs
using DevExpress.Web.Internal;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

public class Device {
    private static BrowserPlatform platform {
        get { return RenderUtils.Browser.Platform; }
    }

    public static bool IsPhone {
        get {
            return platform.IsTouchUI 
                && (platform.IsAndroidMobile || 
                    platform.IsMacOSMobile || 
                    platform.IsWebKitTouchUI || 
                    platform.IsWindowsPhone
                );
        }
    }

    public static bool IsTablet {
        get {
            return IsPhone;
        }
    }

    public static bool IsDesktop {
        get {
            return platform.IsWindows || 
                platform.IsMacOS || 
                platform.IsMSTouchUI;
        }
    }

    private Device() {
    }
}

File: VB/App_Code/Device.vb
Imports DevExpress.Web.Internal
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Public Class Device
    Private Shared ReadOnly Property platform() As BrowserPlatform
        Get
            Return RenderUtils.Browser.Platform
        End Get
    End Property

    Public Shared ReadOnly Property IsPhone() As Boolean
        Get
            Return platform.IsTouchUI AndAlso (platform.IsAndroidMobile OrElse platform.IsMacOSMobile OrElse platform.IsWebKitTouchUI OrElse platform.IsWindowsPhone)
        End Get
    End Property

    Public Shared ReadOnly Property IsTablet() As Boolean
        Get
            Return IsPhone
        End Get
    End Property

    Public Shared ReadOnly Property IsDesktop() As Boolean
        Get
            Return platform.IsWindows OrElse platform.IsMacOS OrElse platform.IsMSTouchUI
        End Get
    End Property

    Private Sub New()
    End Sub
End Class